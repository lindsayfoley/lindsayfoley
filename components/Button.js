import React from "react";
import "../public/styles/_button.scss";

function Button({ link, cta }) {
  return (
    <a href={link}>
      <button>{cta}</button>
    </a>
  );
}

export default Button;
