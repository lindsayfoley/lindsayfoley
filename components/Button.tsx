import React from "react";
import Link from "next/link";
import "../public/styles/Button.scss";

export interface IButtonProps {
  link: string;
  cta: string;
}

const Button: React.FunctionComponent<IButtonProps> = ({ link, cta }) =>
  <Link href={link}>
    <a className="button" itemProp="url">
      {cta}
    </a>
  </Link>;

export default Button;