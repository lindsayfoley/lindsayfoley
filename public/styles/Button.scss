@import "_base.scss";

button {
  width: 250px;
  @include primary-font;
  font-size: 18px;
  font-weight: bold;
  line-height: 35px;
  background: inherit;
  color: $light-color;
  border: $default-border;
  transition: all 0.3s ease 0s;

  &:hover {
    background: $light-color;
    color: $primary-colour;
  }
}