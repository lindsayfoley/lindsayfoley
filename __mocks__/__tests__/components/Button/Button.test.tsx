import React from "react";
import { shallow } from 'enzyme';
import Button, { IButtonProps } from '../../../../components/Button';

const defaultProps: IButtonProps = {
    link: 'test link',
    cta: 'test cta'
}

const setup = (props = {}) => {
    const setupProps = {...defaultProps, ...props}
    return shallow(<Button {...setupProps} />);
}

describe('Button', () => {

    it('renders correctly', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders a button', () => {
        const wrapper = setup();
        expect(wrapper.find('button').length).toBe(1);
        expect(wrapper.find('[href]').length).toBe(1);
        expect(wrapper.prop('href')).toEqual(defaultProps.link);
        expect(wrapper.find('button').text()).toEqual(defaultProps.cta);
    });
});