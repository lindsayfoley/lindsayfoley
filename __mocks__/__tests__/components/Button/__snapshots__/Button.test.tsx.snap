// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<Link
  href="test link"
>
  <a
    className="button"
    itemProp="url"
  >
    test cta
  </a>
</Link>
`;
